jeos-firstboot-enroll

```diff
@@ -52,10 +62,32 @@ enroll_systemd_firstboot() {
 		[ "$result" = 'FIDO2' ] && with_fido2=1
 		[ "$result" = 'TPM2' ] && with_tpm2="$has_tpm2"
 	elif [ "$has_fido2" ]; then
-		dialog $dialog_alternate_screen --backtitle "$PRETTY_NAME" --yesno $"Unlock encrypted disk via FIDO2 token?" 0 0 && with_fido2=1
+		d_styled --yesno $"Unlock encrypted disk via FIDO2 token?" 0 0 && with_fido2=1
 	elif [ -n "$has_tpm2" ]; then
-		dialog $dialog_alternate_screen --backtitle "$PRETTY_NAME" --yesno $"Unlock encrypted disk via TPM?" 0 0 && with_tpm2="$has_tpm2"
+		d_styled --yesno $"Unlock encrypted disk via TPM?" 0 0 && with_tpm2="$has_tpm2"
+	fi
+
+	if [ -n "$with_fido2" ] || [ -n "$with_tpm2" ]; then
+		d_styled --yesno $"Add additional encryption password?" 0 0 || return 0
```

Comment:
If the recovery password is there, why to ask to a lower entropy, less secure password?

There is already a change in cryptenroll that allows the user to use the TPM2 to add/remove a keyslots (like already can be done with the FIDO2 key), but even before that they can use the recover password to add an extra one if it is required.

Reply:
Maybe the password can only be asked iff a TPM2 and FIDO2 key are missing, as a last resort mechanism

Reply:
another idea is to replace the high entropy recovery password with the one in the env var, so the logic would still be consistent: format the luks device with the initial password (automatic or user defined), register the (TPM/FIDO xor user defined password)